### PR TITLE
[REVIEW] Revamp `value_counts` to use groupby count series of any type

### PR DESCRIPTION
**Pull request by galipremsagar**
Resolves: #1951

### CODE REVIEW

File: python/cudf/cudf/dataframe/string.py [hunk 1]
```diff
@@ -795,6 +795,16 @@ def is_monotonic_decreasing(self):
             ).all()
         return self._is_monotonic_decreasing
 
+    def value_counts(self, method="sort"):
+        if method != "sort":
+            msg = "non sort based value_count() not implemented yet"
+            raise NotImplementedError(msg)
```
Comment (kkraus14) [inline python/cudf/cudf/dataframe/string.py]: Lets just remove the method parameter from this function. It's going to use a hash based approach under the hood in this case but it's just confusing.

File: python/cudf/cudf/dataframe/string.py [hunk 2]
```diff
@@ -795,6 +795,16 @@ def is_monotonic_decreasing(self):
...
+        from cudf import Series
+
+        val_counts = Series(self, name=self.name)
+        res = val_counts.groupby(val_counts).count()
+        return res.index, res._column.astype("int64")
```
Comment (kkraus14) [inline python/cudf/cudf/dataframe/string.py]: Any reason we're typecasting to int64 here as opposed to leaving it as an int32?

Comment (galipremsagar) [reply python/cudf/cudf/dataframe/string.py]: matching pandas behavior. pandas `value_counts` return a count series of type `int64`

**Comment by kkraus14** [comment]
@galipremsagar actually, could you extend this PR to just use the groupby based approach under the hood for all data types and remove the method keyword?

**Comment by galipremsagar** [comment]
> @galipremsagar actually, could you extend this PR to just use the groupby based approach under the hood for all data types and remove the method keyword?

@kkraus14 Sure, I'll extend this approach to other types as well.

**Comment by galipremsagar** [comment]
rerun tests

**Comment by galipremsagar** [comment]
rerun tests

File: python/cudf/cudf/dataframe/string.py (hunk 2 above)
Comment (kkraus14) [reply python/cudf/cudf/dataframe/string.py]: Yea understand, but given we've made the decision to limit our dataframe size to int32 rows anyway I think we should just return int32 here for now.

File: python/cudf/cudf/tests/test_dataframe.py [hunk 3]
```diff
@@ -3361,3 +3361,28 @@ def test_one_row_head():
     head_pdf = pdf.head()
 
     assert_eq(head_pdf, head_gdf)
+
+
+def test_value_counts():
```
[the remark this answers is not in the document]
Comment (galipremsagar) [reply python/cudf/cudf/tests/test_dataframe.py]: @beckernick Moved them all to one place.

File: python/cudf/cudf/dataframe/series.py [hunk 4]
```diff
@@ -1818,16 +1818,18 @@ def nunique(self, method="sort", dropna=True):
...
         # return len(self._column.unique())
 
-    def value_counts(self, method="sort", sort=True):
+    def value_counts(self, sort=True):
         """Returns unique values of this Series.
         """
-        if method != "sort":
-            msg = "non sort based value_count() not implemented yet"
-            raise NotImplementedError(msg)
+        from cudf import Series
```
[the remark this answers is not in the document]
Comment (galipremsagar) [reply python/cudf/cudf/dataframe/series.py]: Yikes, not needed actually. Removed it.

**Comment by galipremsagar** [comment]
> Couple of questions. Also, with this refactor, is there any reason we would still need the cudautils function related to value_counts? I believe it only has that singular purpose. If that's the case, we should probably remove it in this PR, too.

Yes, I got rid of the `value_count` kernel aswell in cudautils. We don't use it anywhere else.

File: python/cudf/cudf/dataframe/string.py (hunk 2 above)
Comment (galipremsagar) [reply python/cudf/cudf/dataframe/string.py]: @kkraus14 , removed the explicit typecast to int64.

**Comment by galipremsagar** [comment]
This is ready for review. Once this is merged tests will start passing in #2396

**Pull request merged**